File: ruby/Loot.rb
#encoding: utf-8
require_relative "LootToUI"
#Esta clase representa el botín que se obtiene al vencer a una nave enemiga. 

module Deepspace
	class Loot
		def initialize(supplies, weapons, shields, hangars, medals, ef = false , city = false)
			@nSupplies = supplies
			@nWeapons = weapons
			@nShields = shields
			@nHangars = hangars
			@nMedals = medals
			@getEfficient = ef
			@spaceCity = city
		end
		
			attr_reader :nSupplies , :nWeapons, :nShields, :nHangars , :nMedals, :getEfficient, :spaceCity  	

		def getUIversion
			return LootToUI.new(self)
		end
		def to_s
			getUIversion.to_s
		end
	end
end

File: deepspace/src/deepspace/Loot.java

package deepspace;

public class Loot {
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;

    private boolean getEfficient;
    private boolean spaceCity;
    
    Loot (int nSupplies, int nWeapons, int nShields, int nHangars, int nMedals, boolean ef, boolean city){
        this.nSupplies = nSupplies;
        this.nWeapons = nWeapons;
        this.nShields = nShields;
        this.nHangars = nHangars;
        this.nMedals = nMedals;
        this.getEfficient=ef;
        this.spaceCity=city;
    }
    
    public int getNSupplies(){
        return nSupplies;
    }
    
    public int getNWeapons(){
        return nWeapons;
    } 
    
    public int getNShields(){
        return nShields;
    } 
    
    public int getNHangars(){
        return nHangars;
    } 
    
    public int getNMedals(){
        return nMedals;
    }

    public boolean getEfficient(){
        return getEfficient;
    }

    public boolean spaceCity(){
        return spaceCity;
    }
    
    @Override
    public String toString(){
        String salida="[Loot] -> nSupplies: "+ nSupplies + 
                ", nWeapons: "+ nWeapons +
                ", nShields: "+ nShields +
                ", nHangars: "+ nHangars +
                ", nMedals; "+ nMedals;
        return salida;
    }
    
    LootToUI getUIversion(){
        return (new LootToUI(this));
    }
}

